### PR TITLE
feat: add support for arbitrum & optimism

### PR DESCRIPTION
**Pull request by ujjwal-cyph**
Fixes https://app.clickup.com/t/37308523/CSI-1584

### CODE REVIEW

**Comment by irshadCypherock** [comment]
Task linked: [CU-85zrja2x6 Add support for Arbitrum & Optimism](https://app.clickup.com/t/85zrja2x6)

File: src/constants/coins/eth.ts [hunk 1]
```diff
@@ -129,6 +148,23 @@ export const EthList = [
     chain: 61,
     coinListId: 0xf,
     supportedVersions: [0]
+  }),
+  new EthCoinData({
+    id: EthCoinMap.arbitrum,
+    oldId: 'eth',
```
Comment (irshadCypherock) [inline src/constants/coins/eth.ts]: Make the old Id optional

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: src/constants/coins/eth.ts (hunk 1 above)
Comment (ujjwal-cyph) [reply src/constants/coins/eth.ts]: changed.

File: src/constants/types/AbsCoinData.ts [hunk 2]
```diff
@@ -63,7 +63,7 @@ export abstract class AbsCoinData {
     supportedVersions
   }: IAbsCoinDataOptions) {
     this.id = id;
-    this.oldId = oldId;
+    this.oldId = oldId ?? '';
```
Comment (irshadCypherock) [inline src/constants/types/AbsCoinData.ts]: Is this necessary?

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: src/constants/types/AbsCoinData.ts (hunk 2 above)
Comment (ujjwal-cyph) [reply src/constants/types/AbsCoinData.ts]: no.

Comment (ujjwal-cyph) [reply src/constants/types/AbsCoinData.ts]:
```suggestion
    this.oldId = oldId;
```

Comment (ujjwal-cyph) [reply src/constants/types/AbsCoinData.ts]: Removed

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**